src/Features/VisualBasic/Portable/ReplaceMethodWithProperty/VisualBasicReplaceMethodWithPropertyService.vb

```diff
@@ -0,0 +1,233 @@
+            End If
+
+            Return containingMethod
+        End Function
+
+        Public Sub RemoveSetMethod(editor As SyntaxEditor, setMethodDeclaration As SyntaxNode) Implements IReplaceMethodWithPropertyService.RemoveSetMethod
+            Dim setMethodStatement = TryCast(setMethodDeclaration, MethodStatementSyntax)
+            If setMethodStatement Is Nothing Then
+                Return
+            End If
+
+            Dim methodOrBlock = GetParentIfBlock(setMethodStatement)
+            editor.RemoveNode(methodOrBlock)
+        End Sub
+
+        Public Sub ReplaceGetMethodWithProperty(
+            editor As SyntaxEditor,
+            semanticModel As SemanticModel,
+            getAndSetMethods As GetAndSetMethods,
+            propertyName As String, nameChanged As Boolean) Implements IReplaceMethodWithPropertyService.ReplaceGetMethodWithProperty
+
+            Dim getMethodDeclaration = TryCast(getAndSetMethods.GetMethodDeclaration, MethodStatementSyntax)
+            If getMethodDeclaration Is Nothing Then
+                Return
+            End If
+
+            Dim methodBlockOrStatement = GetParentIfBlock(getMethodDeclaration)
+            editor.ReplaceNode(methodBlockOrStatement,
+                               ConvertMethodsToProperty(editor, semanticModel, getAndSetMethods, propertyName, nameChanged))
+        End Sub
+
+        Private Function GetParentIfBlock(declaration As MethodStatementSyntax) As DeclarationStatementSyntax
+            If declaration.IsParentKind(SyntaxKind.FunctionBlock) OrElse declaration.IsParentKind(SyntaxKind.SubBlock) Then
+                Return DirectCast(declaration.Parent, DeclarationStatementSyntax)
+            End If
+
+            Return declaration
+        End Function
+
+        Private Function ConvertMethodsToProperty(
+            editor As SyntaxEditor,
+            semanticModel As SemanticModel,
+            getAndSetMethods As GetAndSetMethods,
+            propertyName As String, nameChanged As Boolean) As DeclarationStatementSyntax
+
+            Dim generator = editor.Generator
+
+            Dim getMethodStatement = DirectCast(getAndSetMethods.GetMethodDeclaration, MethodStatementSyntax)
+            Dim setMethodStatement = TryCast(getAndSetMethods.SetMethodDeclaration, MethodStatementSyntax)
+
+            Dim propertyNameToken = GetPropertyName(getMethodStatement.Identifier, propertyName, nameChanged)
+
+            Dim newPropertyDeclaration As DeclarationStatementSyntax
+            If getAndSetMethods.SetMethod Is Nothing Then
+                Dim modifiers = getMethodStatement.Modifiers.Add(SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword))
+                Dim propertyStatement = SyntaxFactory.PropertyStatement(
+                        getMethodStatement.AttributeLists, modifiers, propertyNameToken, Nothing,
+                        getMethodStatement.AsClause, initializer:=Nothing, implementsClause:=getMethodStatement.ImplementsClause)
```

Comment:
what about parameters? Don't forget that VB properties can have parameter lists.